addons/sale_stock/report/sale_order_report_templates.xml

```diff
@@ -2,9 +2,9 @@
 <odoo>
     <template id="report_saleorder_document_inherit_sale_stock" inherit_id="sale.report_saleorder_document">
         <xpath expr="//div[@name='expiration_date']" position="after">
-            <div class="col-3" t-if="doc.incoterm" groups="sale_stock.group_display_incoterm">
+            <div class="col-auto col-3 mw-100 mb-2" t-if="doc.incoterm" groups="sale_stock.group_display_incoterm">
```

Comment:
I'm no Bootstrap expert but I believe that `col-auto` has no effect if `col-3` is set.
Could you try removing `col-auto` and see if it works?

Reply:
Hi @AntoineVDV , I had tried the same, but after checking our standard code of sale module(https://github.com/odoo/odoo/blob/master/addons/sale/report/sale_report_templates.xml#L41), i updated the col-auto with col-3.
And actually issue is solved after adding col-auto over there. if there may columns more then 4, this div still occupying width 25% of the block because of 'col-3', not decreasing as per other div.

For better ref i have attached a link,
before this change: https://drive.google.com/file/d/1Jwa8pZk0lWH6gjqK6UJckSHiZ8p_m-Dz/view?usp=sharing

after this change: https://drive.google.com/file/d/1ZC7VrfbT4x2mjhJCpCQHPT9E3wzcdf6w/view?usp=sharing

Hope this will be easy to understand.

Thanks & Regards,

Reply:
> For better ref i have attached a link,
> before this change: https://drive.google.com/file/d/1Jwa8pZk0lWH6gjqK6UJckSHiZ8p_m-Dz/view?usp=sharing
> after this change: https://drive.google.com/file/d/1ZC7VrfbT4x2mjhJCpCQHPT9E3wzcdf6w/view?usp=sharing

I could verify that your fix is indeed working, so no worries about that.

> I had tried the same, but after checking our standard code of sale module(https://github.com/odoo/odoo/blob/master/addons/sale/report/sale_report_templates.xml#L41), i updated the col-auto with col-3.
> And actually issue is solved after adding col-auto over there. if there may columns more then 4, this div still occupying width 25% of the block because of 'col-3', not decreasing as per other div.

This is the part that intrigues me because, from my testing __in a browser__ (not a PDF), the addition of the class `col-3` negates that of the class `col-auto`.
But, again, I'm no Bootstrap expert so maybe that my test was wrong, or maybe that it doesn't behave the same in a PDF generated with wkhtmltodpf, or simply that I tested in the wrong Bootstrap version.

Anyway, if it works and mimics what is done elsewhere, it's fine by me.